File: in_array/in_array.java
import java.util.Arrays;

public class in_array {
    // ATOMIC_FUNCTION:  in_array(value,arr)
    // TAGS: string,list
    // LANG:  java
    // FUNCTION: in_array(value,arr)
    public Boolean in_array(Object value,Object[] arr){
        return Arrays.asList(arr).contains(value);

    }



    public static void main(String args[]) {
        String[] start_val_1 = {"hello","world"};
        String start_val_2 = "hello";
        Boolean expected_end_val=true;
        in_array a= new in_array();

        Boolean end_val = a.in_array(start_val_2,start_val_1);

        // TEST
        if(end_val.equals(expected_end_val)) {
            System.out.println("PASS");
        } else {

            System.out.println("FAIL");
        }
    }
}

File: in_array/in_array.php
<?php
// ATOMIC_FUNCTION:  in_array ( mixed $needle , array $haystack [, bool $strict = FALSE ] ) 
// TAGS: array, Boolean
// LANG:  php
// FUNCTION: in_array ( mixed $needle , array $haystack [, bool $strict = FALSE ] ) 

$start_val_1 = array("hello","world");
$start_val_2 = "hello";
$expected_end_val = True;

$end_val = in_array($start_val_2,$start_val_1);

// TEST
if($end_val == $expected_end_val) {
	echo "PASS";
} else {
	echo "FAIL";
}

?>
